vector/src/main/java/im/vector/app/features/home/room/detail/timeline/TimelineEventController.kt

```diff
@@ -185,17 +186,21 @@ class TimelineEventController @Inject constructor(private val dateFormatter: Vec
         override fun onChanged(position: Int, count: Int, payload: Any?) {
             synchronized(modelCache) {
                 assertUpdateCallbacksAllowed()
+                Timber.v("listUpdateCallback.onChanged(position: $position, count: $count). " +
+                        "\ncurrentSnapshot has size of ${currentSnapshot.size} items")
                 (position until position + count).forEach {
                     // Invalidate cache
                     modelCache[it] = null
                 }
                 // Also invalidate the first previous displayable event if
                 // it's sent by the same user so we are sure we have up to date information.
                 val invalidatedSenderId: String? = currentSnapshot.getOrNull(position)?.senderInfo?.userId
-                val prevDisplayableEventIndex = currentSnapshot.subList(0, position).indexOfLast {
+                // In some cases onChanged will be called before onRemoved and onInserted so position will be smaller than currentSnapshot.size.
+                val prevList = currentSnapshot.subList(0, min(position, currentSnapshot.size))
+                val prevDisplayableEventIndex = prevList.indexOfLast {
                     timelineEventVisibilityHelper.shouldShowEvent(it, partialState.highlightedEventId)
                 }
-                if (prevDisplayableEventIndex != -1 && currentSnapshot[prevDisplayableEventIndex].senderInfo.userId == invalidatedSenderId) {
+                if (prevDisplayableEventIndex != -1 && currentSnapshot.getOrNull(prevDisplayableEventIndex)?.senderInfo?.userId == invalidatedSenderId) {
```

Comment:
I do not think this change is necessary, since `prevDisplayableEventIndex` will always be in the range `[0, currentSnapshot.size[`.
Anyway, better safe than sorry I guess.